### ISSUE TITLE
Add reactive streams in published ZIP file as a dependency, for convenience

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#9329]
jOOQ 3.12 added a dependency on reactive streams. This dependency is missing when downloading jOOQ from our website, which leads to frustrating error messages. Users have to download that dependency manually. Other dependencies are optional and don't fail early unless users need them (e.g. JPA annotations), but this one leads to

```
java.lang.NoClassDefFoundError: org/reactivestreams/Publisher
```

When starting using jOOQ. We should include the dependency in the ZIP file downloadable from our website. It is not a problem when using Maven, in case of which the dependency is resolved automatically.

The license is public domain, which is compatible with our ASL 2.0 license:
https://github.com/reactive-streams/reactive-streams-jvm/blob/master/LICENSE

**Comment by ByRicardoSimoes** [comment]
For users that don't quite get how to work around it:
Follow the tutorial, but change the command to this:
java -cp jooq-3.12.1.jar;jooq-meta-3.12.1.jar;jooq-codegen-3.12.1.jar;reactive-streams-1.0.2.jar;mysql-connector-java-8.0.17.jar org.jooq.codegen.GenerationTool library.xml

ofcourse, youll need all the .jars listed above. I have not tested it with the normal 3 jooq jars + jdbxc jar and reactive jar, but that might also work.

**Comment by lukaseder** [comment]
Thanks for documenting this here!

**Comment by lukaseder** [comment]
We're currently building the ZIP file with ant, after a Maven build. This means we can assume that the dependency has been downloaded to the local `~/.m2` repository by Maven, and copy it from there.

However, if we had some synergies between this task, and a planned snapshot repository, that might be even better.

**Comment by lukaseder** [comment]
Implemented in 3.13.0. Backport scheduled for 3.12.3 (#9452)
